lib/plugins/aws/package/lib/mergeIamTemplates.js

```diff
@@ -37,7 +37,14 @@ module.exports = {
         );
       });
 
+    const { iam } = this.serverless.service.provider;
+    const iamRole = iam && iam.role ? iam.role : {};
+
     // resolve early if provider level role is provided
+    if (typeof iamRole === 'string') {
+      return;
+    }
+
     if ('role' in this.serverless.service.provider) {
```

Comment:
I think we should show deprecation here as well (?)

Reply:
Sure, I was expecting that all (or most?) usage will result in calling code from `provider.getCustomExecutionRole()` that does report the deprecation.